zerver/tests/test_user_topics.py

```diff
@@ -229,6 +232,225 @@ def test_muted_topic_remove_invalid(self) -> None:
         self.assert_json_error(result, "Please choose one: 'stream' or 'stream_id'.")
 
 
+class MutedTopicsTests(ZulipTestCase):
+    def test_get_deactivated_muted_topic(self) -> None:
+        user = self.example_user("hamlet")
+        self.login_user(user)
+
+        stream = get_stream("Verona", user.realm)
+
+        url = "/api/v1/user_topics"
+        data = {
+            "stream_id": stream.id,
+            "topic": "Verona3",
+            "visibility_policy": UserTopic.VisibilityPolicy.MUTED,
+        }
+
+        mock_date_muted = datetime(2020, 1, 1, tzinfo=timezone.utc).timestamp()
+        with time_machine.travel(datetime(2020, 1, 1, tzinfo=timezone.utc), tick=False):
+            result = self.api_post(user, url, data)
+            self.assert_json_success(result)
+
+        stream.deactivated = True
+        stream.save()
+
+        self.assertNotIn((stream.name, "Verona3", mock_date_muted), get_topic_mutes(user))
+        self.assertIn((stream.name, "Verona3", mock_date_muted), get_topic_mutes(user, True))
+
+    def test_user_ids_muting_topic(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        realm = hamlet.realm
+        stream = get_stream("Verona", realm)
+        topic_name = "teST topic"
+        date_muted = datetime(2020, 1, 1, tzinfo=timezone.utc)
+
+        stream_topic_target = StreamTopicTarget(
+            stream_id=stream.id,
+            topic_name=topic_name,
+        )
+
+        user_ids = stream_topic_target.user_ids_with_visibility_policy(
+            UserTopic.VisibilityPolicy.MUTED
+        )
+        self.assertEqual(user_ids, set())
+
+        url = "/api/v1/user_topics"
+
+        def set_topic_visibility_for_user(user: UserProfile, visibility_policy: int) -> None:
+            data = {
+                "stream_id": stream.id,
+                "topic": "test TOPIC",
+                "visibility_policy": visibility_policy,
+            }
+            with time_machine.travel(datetime(2020, 1, 1, tzinfo=timezone.utc), tick=False):
```

Comment:
Let's use the `date_muted` variable in this line -- it's best not to repeat long timestamps like this one.

Reply:
Makes sense! -- updated.